lib/osf-components/addon/components/file-browser/add-new/upload/component.ts

```diff
@@ -0,0 +1,121 @@
+import { action, notifyPropertyChange } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+import OsfStorageManager from 'osf-components/components/storage-provider-manager/osf-storage-manager/component';
+import { taskFor } from 'ember-concurrency-ts';
+
+interface Args {
+    manager: OsfStorageManager;
+    isOpen: boolean;
+}
+
+export default class Upload extends Component<Args> {
+    dropzoneOptions = {
+        createImageThumbnails: false,
+        method: 'PUT',
+        withCredentials: true,
+        preventMultipleFiles: false,
+        acceptDirectories: false,
+        autoProcessQueue: true,
+    };
+
+    @service intl!: Intl;
+    @service toast!: Toast;
+    @tracked uploading: any[] = [];
+    @tracked uploadCompleted: any[] = [];
+    @tracked uploadErrored: any[] = [];
+    @tracked clickableElementId = '';
+
+    get clickableElementSelectors() {
+        if (this.clickableElementId) {
+            return [`#${this.clickableElementId}`];
+        }
+        return [];
+    }
+
+    get shouldShowUploadingModal() {
+        return this.uploading.length !== 0;
+    }
+
+    get shouldShowFailureModal() {
+        return this.uploadErrored.length !== 0 &&
+            this.uploading.length === 0;
+    }
+
+    get shouldShowSuccessModal() {
+        return this.uploadCompleted.length !== 0 &&
+            this.uploadErrored.length === 0 &&
+            this.uploading.length === 0;
+    }
+
+    get shouldShowModal() {
+        return this.shouldShowFailureModal ||
+            this.shouldShowUploadingModal ||
+            this.shouldShowSuccessModal;
+    }
+
+    @action
+    buildUrl(files: any[]) {
+        const { name, newUploadLink } = files[0];
+        if (newUploadLink) {
+            return newUploadLink;
+        }
+        return `${this.args.manager.currentFolder.links.upload}?${$.param({ kind: 'file', name })}`;
```

Comment:
I don't think this is necessary right now, but we can future-proof this by using URL.searchParams